low-level/files/opt/seagate/sspl/bin/sspl_setup_init.py

```diff
@@ -0,0 +1,174 @@
+            # No processes to check in vm env
+        else:
+            print(f"No rpm or process dependencies set, to check for supplied role {role}, skipping checks.\n")
+
+    def get_uid(self, proc_name : str) -> int:
+        uid = -1
+        try :
+            uid = pwd.getpwnam(proc_name).pw_uid
+        except KeyError :
+            print(f"No such User: {proc_name}")
+        return uid
+
+    def process(self):
+        if not self.args:
+            self.usage()
+
+        i =0
+        while i < len(self.args):
+            if self.args[i] == '-dp':
+                # Extract the data path
+                sspldp = ''
+                with open(file_store_config_path, mode='rt') as confile:
+                    for line in confile:
+                        if line.find('data_path') != -1:
+                            sspldp = line.split('=')[1]
+                            break
+                # Crete the directory and assign permissions
+                try:
+                    # what should be the permissions for this directory ??
+                    os.makedirs(sspldp, mode=0o766, exist_ok=True)
+                    sspl_ll_uid = self.get_uid('sspl-ll')
+                    if sspl_ll_uid == -1:
+                        sys.exit(1)
+                    os.chown(sspldp, sspl_ll_uid, -1)
+                    for root, dirs, files in os.walk(sspldp):
+                        for item in dirs:
+                            os.chown(os.path.join(root, item), sspl_ll_uid, -1)
+                        for item in files:
+                            os.chown(os.path.join(root, item), sspl_ll_uid, -1)
+                except OSError as error:
+                    print(error) 
+
+            elif self.args[i] == '-r':
+                i+=1
+                if i>= len(self.args) or self.args[i] not in roles:
+                    self.usage()
+                else:
+                    self.role = self.args[i]
+            else:
+                self.usage()
+            i+=1
+        
+        # Create /tmp/dcs/hpi if required. Not needed for '<product>' role
+        if self.role != "cortx":
+            try:
+                os.makedirs('/tmp/dcs/hpi', mode=0o777, exist_ok=True)
+                zabbix_uid = self.get_uid('zabbix')
+                if zabbix_uid != -1:
+                    os.chown('/tmp/dcs/hpi', zabbix_uid, -1)
+            except OSError as error:
+                print(error) 
+
+        # Check for sspl required processes and misc dependencies like installation, etc based on 'role'
+        if self.role : 
+            self.check_dependencies(self.role)
+        
+        # Create mdadm.conf to set ACL on it.
+        with open('/etc/mdadm.conf', 'a'):
+            os.utime('/etc/mdadm.conf')
+
+        # self._send_command('setfacl -m u:sspl-ll:rw /etc/mdadm.conf') 
+        #  What should be the file permissions need here and are `setfacl` and below impl equivalent??
```

Comment:
Please address this.

Reply:
file permission, again read write for all users, its not user specific.
but for setfacl i dont see easy python alternative, you check more still. In this case though feel use sub-process to execute setfacl here.

Reply:
subprocess call for setfacl gives error "Operation not supported" need to dig if something is possible.

Reply:
can you try as root user